agent/dns.go

```diff
@@ -567,7 +567,7 @@ func trimUDPResponse(config *DNSConfig, req, resp *dns.Msg) (trimmed bool) {
 
 	// This cuts UDP responses to a useful but limited number of responses.
 	maxAnswers := lib.MinInt(maxUDPAnswerLimit, config.UDPAnswerLimit)
-	if maxSize == defaultMaxUDPSize && numAnswers > maxAnswers {
+	if numAnswers > maxAnswers {
```

Comment:
I think the idea here is that, if request is edns and the size has changed from the default max size, we don't want to trim the answers or else they will be truncated.

Reply:
@sethvargo  I'm not sure to understand what you mean. Would you have an example?

Reply:
I think you can try "+bufsize=512" with dig , because the dig may not build with edns.